superset-frontend/src/dashboard/components/menu/ShareMenuItems/index.tsx

```diff
@@ -49,23 +45,17 @@ const ShareMenuItems = (props: ShareMenuItemProps) => {
     addDangerToast,
     addSuccessToast,
     dashboardId,
-    formData,
+    hash,
     ...rest
   } = props;
 
   async function generateUrl() {
-    // chart
-    if (formData) {
-      // we need to remove reserved dashboard url params
-      return getChartPermalink(formData, RESERVED_DASHBOARD_URL_PARAMS);
-    }
-    // dashboard
     const nativeFiltersKey = getUrlParam(URL_PARAMS.nativeFiltersKey);
     let filterState = {};
     if (nativeFiltersKey && dashboardId) {
       filterState = await getFilterValue(dashboardId, nativeFiltersKey);
     }
-    return getDashboardPermalink(String(dashboardId), filterState);
+    return getDashboardPermalink(String(dashboardId), filterState, hash);
```

Comment:
Would be nice if we make this more explicit:

```suggestion
    return getDashboardPermalink({
      dashboardId,   # need to relax the type restriction in `getDashboardPermalink` as well
      filterState,
      hash: dashboardComponentId
    });
```